src/software/thunderscope/wifi_communication_manager.py

```diff
@@ -43,12 +50,12 @@ def __init__(
         """
         ## Robot IP address tracking ##
         self.robot_ip_addresses: list[tuple[Lock, str | None]] = [
-            (Lock(), None) for _ in range(MAX_ROBOT_IDS_PER_SIDE)
+            (Lock(), None) for ip in robot_ip
         ]
 
         ## Senders and listeners ##
         self.primitive_senders: list[tuple[Lock, tbots_cpp.PrimitiveSender | None]] = [
-            (Lock(), None) for _ in range(MAX_ROBOT_IDS_PER_SIDE)
+            (Lock(), None) for ip in robot_ip
```

Comment:
I believe you can leave this as it was before. The code at some point will look at `self.robot_ip_addresses` and set this structure up.